Lecture2/src/task4/view/Menu.java

```diff
@@ -0,0 +1,82 @@
+package task4.view;
+
+import task4.model.MarathonParticipant;
+import task4.util.MarathonHelper;
+
+import java.util.List;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Menu {
+
+    private static Scanner scanner = new Scanner(System.in);
+    private String menu;
+
+    public void printMenu(List<MarathonParticipant> participants) {
+        do {
+            printSeparator();
+            System.out.println("1 - Print all participants");
+            System.out.println("2 - Print participant by place");
+            System.out.println("q - Exit");
+            menu = readMenu();
+            switch (menu) {
+                case "1": {
+                    MarathonHelper.printParticipants(participants);
+                    break;
+                }
+                case "2": {
+                    printByPlace(participants);
+                    break;
+                }
+                case "q":{
+                    break;
+                }
+            }
+        } while (!menu.equals("q"));
+    }
+
+    private void printByPlace(List<MarathonParticipant> participants) {
+        MarathonHelper.printParticipantByPlace(participants, readPlace(participants.size()));
+    }
+
+    private static boolean isMenu(String string) {
+        Pattern p = Pattern.compile("^[a-z0-9]{1,1}$");
+        Matcher m = p.matcher(string);
+        return m.matches();
+    }
+
+    private static String readMenu() {
+        String string = "";
+        while (true) {
+            System.out.print("Choose: ");
+            string = scanner.next();
+            if (isMenu(string)) {
+                break;
+            }
+            System.out.println("You entered incorrect data!");
+        }
+        return string;
+    }
+
+    private static int readPlace(int maxPlace) {
+        int place = 0;
+        do {
+            System.out.print("Enter place: ");
+            while (!scanner.hasNextInt()) {
+                scanner.next();
+                System.out.println("You entered not a number!");
+            }
+            place = scanner.nextInt();
+            if (place < maxPlace) {
```

Comment:
0 place it is incorrect program behavior

Reply:
fix